aten/src/ATen/DLConvertor.cpp

```diff
@@ -227,12 +228,23 @@ DLManagedTensor* toDLPack(const Tensor& src) {
   atDLMTensor->tensor.dl_tensor.device = getDLDevice(src, device_id);
   atDLMTensor->tensor.dl_tensor.ndim = src.dim();
   atDLMTensor->tensor.dl_tensor.dtype = getDLDataType(src);
+  // Normalize the strides to 1 wherever shape < 2
+  auto shape = src.sizes().data();
+  int64_t *strides = new int64_t[src.dim()];
```

Comment:
Do you guys have something like a `MAX_DIM` concept? perhaps you could add a slot `strides[MAX_DIM]` in the `ATenDLMTensor` structure to store the strides. That way you can implement this hole thing with a single `new` allocation. 

```c++
int64_t *strides = ATenDLMTensor->strides;
```

PS: This is just a suggestion for an alternative implementation. I'm totally fine with things as they are now.

Reply:
There is a [concept of MAX_DIMS](https://github.com/pytorch/pytorch/blob/86de9e72914f970556144f64fb7c779e32a41005/torch/csrc/utils/tensor_new.cpp#L54) but I am not sure the `src.sizes.data()` is forced to be a `int64_t[MAX_DIMS]`. If it is shorter, arbitrary memory would be copied into the `ATenDLMTensor.strides`.

Reply:
> I am not sure the `src.sizes.data()` is forced to be a `int64_t[MAX_DIMS]`

It does not have to be. It does not matter.

> If it is shorter, arbitrary memory would be copied into the `ATenDLMTensor.strides`.

No, simply because the `for` loop below fills up to `src.dim()` entries in the `stride` buffer. Any entries beyond `src.dim()` will be unused.

The real issue is whether you are willing to pay the extra 1 KiB needed to store the `MAX_DIMS` strides vs saving one `new` allocation call. 

Am I making myself clear enough?

Reply:
Ahh right. I misunderstood. I will let other reviewers weigh in whether they want to avoid the new call.

Reply:
I don't really care about perf here, I just want to minimize manual memory management. Probably the slickest way to do this is to create a *new* Tensor that has the normalized strides and then use the fields on that tensor directly from dlpack. WDYT

Reply:
Is there an API to create a Tensor copy that shares data but not metadata?

Reply:
yeah, `as_strided` is the ticket here

Reply:
In order to use `as_strided` I need to allocate an int vector for the new strides, right? I am not sure what advantage creating a new container gives over using the existing capsule deallocate mechanism.

Reply:
As I said, the benefit is you don't have to manually memory manage strides separately from Tensor object. Tensor is the owning object that is managed and everything else hangs off it.

Reply:
Done (assuming CI passes)